### PR TITLE
Un special-case system drive btrfs-in-partition treatment #2824

### PR DESCRIPTION
**Pull request by phillxnet**
No functional change on first commit. Some basic code improvements to clear the way for the planed simplification/modification later in this draft PR.

Includes:
- Modernisation re type hints in scan_disks().
- Programmatically establish `lsblk` field derived Disk namedtuple fields.
- Refactoring for readability.
- Black reformatting.

See issue text #2824

### CODE REVIEW

File: src/rockstor/system/osi.py [hunk 1]
```diff
@@ -108,12 +108,25 @@
     "0123456789ABCDEF",  # No-name USB external multi-bay.
 ]
 
-Disk = collections.namedtuple(
-    "Disk",
-    "name model serial size transport vendor "
-    "hctl type fstype label uuid parted root "
-    "partitions",
-)
+# Requried to assist parsing of lsblk output to Disk field names.
```
Comment (Hooverdan96) [inline src/rockstor/system/osi.py:111]: Small typo:
# Required to assist parsing of lsblk output to Disk field names.

File: src/rockstor/system/osi.py [hunk 2]
```diff
@@ -265,7 +278,20 @@ def run_command(
     return out, err, rc
 
 
-def scan_disks(min_size, test_mode=False):
+# def lsblk_line_parse(line: str = "") -> Disk:
+#     """
+#     Parse single lsblk output line into a Disk named tupile
```
Comment (Hooverdan96) [inline src/rockstor/system/osi.py:283]: small typo:
```
# Parse single lsblk output line into a Disk named tuple
```

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for the review: but this is very much a draft still: I've not actually begun the functional changes. And I'll have to squash these and many other changes to come so your contributions/attribution will lost.

But thanks for the typos anyway. I'll incorporate when next working on this draft: soon hopefully. Pushing my work as I go here though just in case - and easier to ask advice etc if I get stuck. Or do end-to-end testing via our back-end rpmbuild against PRs (draft or otherwise). I think I have a few simplifications on the way, and hopefully the end result will be more maintainable. Pretty sure we can simplify quite a bit: as what we have currently is definitely super stretched. I also plan to make more use of our Disk namedtuple in scan_disks() so the code can be more self documenting, and to help guard against accidental changes we don't intend: named tuples are immutable.

**Comment by Hooverdan96** [comment]
Understood absolutely, and I'm not craving attribution for those cosmetic things, but figured if I mention them now, you can get rid of that, especially since you will now start focusing on making actual functional changes...

Godspeed on the simplification, and definitely makes sense to me if you will be able to put a more complete dataset/logic on the `scan_disks()` to simplify subsequent functions and overall usage for other pieces.

**Comment by phillxnet** [comment]
At this point in this draft PR associated branch all existing osi.py tests again pass. And we have maintained the existing observed failure in the 2 new tests associated with draft PR: "Disk being misattributed to ROOT pool #2749" #2823.

So we are still at no functional change: as intended. But the code is now likely more robust and easier to read with many more type hints and way more obvious re internal changes and intent within scan_disks(). Next:

- [ ]  a little more tidy, and give the increased readability we can likely do away with many of the explanatory comments to further readability.
- [ ] modify all existing tests to expect the issue associated behaviour change.
- [ ] approach the functional changes in scan_disks() to return all tests to passing again.
- [ ] ensure we have accounted for and fixed the failure detailed in the 2 new tests associated with #2749 & #2823

**Comment by phillxnet** [comment]
This Draft PR's work-to-date represents a significant enough modification to address a stepping stone defined by the following issue: Modernise scan_disks() - no functional change intended #2826 for which a new PR will be presented shorlty.

**Pull request closed without merging**